src/main/scala/fs2/Process1.scala

```diff
@@ -18,6 +18,20 @@ object process1 {
   def chunks[F[_],I](implicit F: NotNothing[F]): Handle[F,I] => Pull[F,Chunk[I],Handle[F,I]] =
     h => h.await flatMap { case chunk #: h => Pull.output1(chunk) >> chunks.apply(h) }
 
+  /** Emit inputs which match the supplied predicate to the output of the returned `Pull` */
+  def filter[F[_], I](f: I => Boolean)(implicit F: NotNothing[F]): Handle[F,I] => Pull[F,I,Handle[F,I]] =
+    h => h.await flatMap { case chunk #: h =>
+      val filtered =
```

Comment:
I'd suggest just adding a `filter` method to `Chunk`, which can be overridden with some more efficient implementation in the various `Chunk` types. I definitely would not include the special-casing here.

Then your implementation just becomes `h.await flatMap { case chunk #: h => Pull.output(chunk filter f) >> filter(f).apply(h) }`